docs/reference/fields.txt

```diff
@@ -471,6 +471,93 @@ matches strings containing "hello" before a newline, besides strings ending in
 This is because the meaning of ``$`` is different between PCRE and Ruby
 regular expressions.
 
+
+.. _typed-arrays:
+
+Field Type: Typed Arrays
+------------------------
+
+Mongoid supports declaring a typed array field on models. This field will
+enforce the type of the elements in the array by ``demongoizing`` each element
+inserted into the array. To make a typed array field:
+
+.. code:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :members, type: Array(String)
+  end
+
+The members field on the Band class will have class ``Mongoid::StringArray``.
```

Comment:
I would prefer the class nested under Fields, e.g. `Mongoid::Fields::StringArray` or in a dedicated module like `Mongoid::Fields::TypedArray::StringArray`.

Reply:
mmm alright, but for the record, I think `Mongoid::StringArray` is super clean and nice looking.